src/rtutils.c

```diff
@@ -804,9 +835,7 @@ static size_t jl_static_show_x_(JL_STREAM *out, jl_value_t *v, jl_datatype_t *vt
                 n += jl_printf(out, ")");
             }
         }
-        if (dv->parameters && (jl_value_t*)dv != dv->name->wrapper &&
-            (jl_has_free_typevars(v) ||
-             (jl_value_t*)dv != (jl_value_t*)jl_tuple_type)) {
```

Comment:
I see the `jl_tuple_type` check was to make sure `Tuple` is printed as `Tuple`, but I don't know how the free_typevars check makes sense :shrug: